ocaml/xapi/helpers.ml

```diff
@@ -1153,3 +1153,16 @@ let force_loopback_vbd ~__context =
 	let other_config = Db.Pool.get_other_config ~__context ~self:pool in
 	List.mem_assoc "force_loopback_vbd" other_config
 
+let assert_virt_hw_support ~__context ~vm_virt_hw_vn ~host_to =
+	let host_virt_hw_vns = 
+	try
+		match host_to with
+        | LocalObject host_ref ->
+        	Db.Host.get_virt_hw_vns ~__context ~self:host_ref
+        | RemoteObject (rpc, session_id, host_ref) ->
+        	Client.Client.Host.get_virt_hw_vns ~rpc ~session_id ~self:host_ref
+    with Not_found ->
+    	[] in
+	
+	if not (List.exists (fun virt_hw_vn -> virt_hw_vn = vm_virt_hw_vn) host_virt_hw_vns) then
```

Comment:
Better to use `List.mem` here:
`if not (List.mem vm_virt_hw_vn host_virt_hw_vns) then ...`
and in the Server_error it would be nice to specify the version needed by the VM.